test/integration/singlecluster/controller/failurerecovery/pod_termination_controller_test.go

```diff
@@ -0,0 +1,102 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package failurerecovery
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	testingnode "sigs.k8s.io/kueue/pkg/util/testingjobs/node"
+	testingpod "sigs.k8s.io/kueue/pkg/util/testingjobs/pod"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+const (
+	unreachableNodeName = "unreachable-node"
+	reachableNodeName   = "reachable-node"
+)
+
+func createTerminatingPod(p *corev1.Pod) {
+	ginkgo.GinkgoHelper()
+
+	util.MustCreate(ctx, k8sClient, p)
+	gomega.ExpectWithOffset(1, k8sClient.Delete(ctx, p)).To(gomega.Succeed())
+}
+
+var _ = ginkgo.Describe("Pod termination controller", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	var ns *corev1.Namespace
+	var matchingPodWrapper *testingpod.PodWrapper
+
+	ginkgo.BeforeAll(func() {
+		ns = util.CreateNamespaceFromPrefixWithLog(ctx, k8sClient, "pod-fr-namespace-")
+		nodes := []corev1.Node{
+			*testingnode.MakeNode(unreachableNodeName).
+				NotReady().
+				Taints(corev1.Taint{Key: corev1.TaintNodeUnreachable, Effect: corev1.TaintEffectNoSchedule}).
+				Obj(),
+			*testingnode.MakeNode(reachableNodeName).
+				Ready().
+				Obj(),
+		}
+		util.CreateNodesWithStatus(ctx, k8sClient, nodes)
+
+		matchingPodWrapper = testingpod.MakePod("matching-pod", ns.Name).
+			StatusPhase(corev1.PodPending).
+			TerminationGracePeriod(1).
+			NodeName(unreachableNodeName).
+			Annotation("kueue.x-k8s.io/safe-to-forcefully-terminate", "true")
+	})
+
+	ginkgo.It("forcefully terminates pods that opt-in, scheduled on unreachable nodes", func() {
+		matchingPod := matchingPodWrapper.Clone().Obj()
+		createTerminatingPod(matchingPod)
+
+		gomega.Eventually(func(g gomega.Gomega) {
+			g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: matchingPod.Name, Namespace: matchingPod.Namespace}, matchingPod)).
+				To(gomega.Succeed())
+			g.Expect(matchingPod.Status.Phase).Should(gomega.Equal(corev1.PodFailed))
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+	})
+
+	ginkgo.It("does not forcefully terminate pods that did not opt-in, scheduled on unreachable nodes", func() {
+		nonMatchingPod := matchingPodWrapper.
+			Clone().
+			Name("non-matching-pod").
+			Annotation("kueue.x-k8s.io/safe-to-forcefully-terminate", "false").
+			Obj()
+		createTerminatingPod(nonMatchingPod)
+
+		gomega.Consistently(func(g gomega.Gomega) {
+			g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: nonMatchingPod.Name, Namespace: nonMatchingPod.Namespace}, nonMatchingPod)).
+				To(gomega.Succeed())
+			g.Expect(nonMatchingPod.Status.Phase).Should(gomega.Equal(corev1.PodPending))
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
Use util.ConsistentDuration for consistency checks. util.Timeout is 10s which would be too much time added.

Reply:
I see, using 2s for Consistency check is fine.